crates/analyzer/tests/cases/flow_php_integration.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace Flow\Types {
+    /**
+     * @template T
+     */
+    interface Type
+    {
+        /**
+         * @param mixed $value
+         *
+         * @return T
+         */
+        public function assert($value): mixed;
+    }
+}
+
+namespace Flow\Types\DSL {
+    use Flow\Types\Type;
+
+    /**
+     * @template T
+     *
+     * @param array<string, Type<T>> $elements
+     *
+     * @return Type<array<string, T>>
+     */
+    function type_structure(array $elements = [], array $optional_elements = [], bool $allow_extra = false): Type
+    {
+        return type_structure($elements, $optional_elements, $allow_extra);
+    }
+
+    /**
+     * @return Type<string>
+     */
+    function type_string(): Type
+    {
+        return type_string();
+    }
+
+    /**
+     * @return Type<int>
+     */
+    function type_integer(): Type
+    {
+        return type_integer();
+    }
+
+    /**
+     * @template T
+     *
+     * @param Type<T> $element
+     *
+     * @return Type<list<T>>
+     */
+    function type_list(Type $element): Type
+    {
+        return type_list($element);
+    }
+
+    /**
+     * @template T of object
+     *
+     * @param class-string<T> $class
+     *
+     * @return Type<T>
+     */
+    function type_instance_of(string $class): Type
+    {
+        return type_instance_of($class);
+    }
+}
+
+namespace {
+    function get_mixed(): mixed
+    {
+        return 1;
+    }
+
+    function i_take_string(string $value): void
+    {
+        echo "Received string: $value\n";
+    }
+
+    function i_take_int(int $value): void
+    {
+        echo "Received int: $value\n";
+    }
+
+    $array_type = Flow\Types\DSL\type_structure([
+        'name' => Flow\Types\DSL\type_string(),
+        'age' => Flow\Types\DSL\type_integer(),
+        'address' => Flow\Types\DSL\type_structure([
+            'street' => Flow\Types\DSL\type_string(),
+            'city' => Flow\Types\DSL\type_string(),
+        ], [
+            'country' => Flow\Types\DSL\type_string(),
+        ]),
+    ]);
+
+    $flexible_type = Flow\Types\DSL\type_structure(
+        [
+            'required_field' => Flow\Types\DSL\type_string(),
+        ],
+        [],
+        true,
+    );
+
+    $array = $array_type->assert(get_mixed());
+    $flexible = $flexible_type->assert(get_mixed());
+
+    /* @mago-expect analysis:type-confirmation */
+    Mago\confirm($flexible, 'array{required_field: string, ...}');
```

Comment:
This actually does do anything, Mago will trigger `type-confirmation` issue regardless if the type matches or not.. we should probably add a utility that triggers only on failure.

However, for now, i suggest creating a function that expects `array{required_field: string, ...}` and passing it as an argument, which should yield no errors.

Reply:
thanks, i'll take a look

Reply:
@azjezz please take another look.
added
```php
    /**
     * @param array{required_field: string, ...} $value
     * @return array{required_field: string, ...}
     */
    function i_take_flexible_array(array $value): array
    {
        return $value;
    }
```

which i think is what you mean?